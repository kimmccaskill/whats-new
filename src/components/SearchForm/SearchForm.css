.nav-bar {
  padding: 2% 0 2% 0;
  display: flex;
}

.search-input {
  width: 50%;
  height: 50px;
  font-size: 1em;
  padding: 1%;
  border-radius: 30px;
}

.search-btn {
  border-radius: 30px;
  margin-left: 10px;
  width: 20%;
  font-size: 1em;
}
