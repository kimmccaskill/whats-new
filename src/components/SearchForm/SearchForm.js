import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
  constructor(props) {
    super(props);
    this.state = {searchInput: ''}
  }

  updateChange = (event) => {
    this.setState({searchInput: event.target.value})
  }

  submitSearch = () => {
    this.props.filterArticles(this.state.searchInput)
  }

  render() {
    return (
      <form className="nav-bar">
        <input
          className="search-input"
          type="text"
          placeholder="Search for article here"
          value={this.state.searchInput}
          onChange={this.updateChange}
        />
        <button
          type="button"
          className="search-btn"
          onClick={this.submitSearch}
        >Search</button>
      </form>
    )
  }
}

export default SearchForm;
