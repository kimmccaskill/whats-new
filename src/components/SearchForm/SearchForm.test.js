import React from 'react';
import ReactDOM from 'react-dom';
import SearchForm from './SearchForm';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  const mockFilterArticles = jest.fn();
  beforeEach(() => {

    wrapper = shallow(<SearchForm filterArticles={mockFilterArticles}/>);
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state of no ideas', () => {
  expect(wrapper.state('searchInput')).toEqual('')
  });

  it('should update state with updateChange method', () => {
  const mockEvent = {
    target: {
      value: 'birdcall'
    }
  }

  wrapper.instance().updateChange(mockEvent);

  expect(wrapper.state('searchInput')).toEqual('birdcall');
  });
});
