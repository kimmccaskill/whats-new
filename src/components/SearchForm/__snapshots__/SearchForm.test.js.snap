// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<form
  className="nav-bar"
>
  <input
    className="search-input"
    onChange={[Function]}
    placeholder="Search for article here"
    type="text"
    value=""
  />
  <button
    className="search-btn"
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
</form>
`;
